### PR TITLE
Checklisting – add authentication

### LINKED ISSUE #2300: Checklisting – Authentication
Currently anyone can edit any checklist. This needs to change before the feature goes anywhere.

### PR DESCRIPTION
**Pull request by lmcnulty**
Resolves #2300.

### CODE REVIEW

File: site/gatsby-site/src/pages/apps/checklists.js [hunk 1]
```diff
@@ -95,29 +99,29 @@ const ChecklistsPageBody = ({ taxa, classifications, t }) => {
...
-          savedChecklist || {
+          savedChecklist /* || {
             name: 'Unspecified System',
             about: '',
             tags_goals: [],
             tags_methods: [],
             tags_other: [],
             risks: [],
-          }
+          } */
```
Comment (pdcp1) [inline site/gatsby-site/src/pages/apps/checklists.js:121]: We should delete this commented code if it's not needed

File: site/gatsby-site/src/graphql/checklists.js [hunk 2]
```diff
@@ -53,6 +54,14 @@ export const UPDATE_CHECKLIST = gql`
...
 `;
 
+export const INSERT_EMPTY_CHECKLIST = gql`
+  mutation insertChecklist {
+    insertOneChecklist {
+      _id
+    }
+  }
+`;
+
```
Comment (pdcp1) [inline site/gatsby-site/src/graphql/checklists.js:64]: I didn't find where is this mutation being used. Please delete this declaration statement if it's not used.

**Review by pdcp1 — CHANGES_REQUESTED** [review]
Please add some tests for all possible cases (ie: authenticated as owner, authenticated as not the owner, unauthenticated)

**Comment by pdcp1** [comment]
@lmcnulty do you know why the build is failing? It should pass before merge this.

**Comment by lmcnulty** [comment]
@pdcp1 It's building now, I had to update a line a rebuild after a test flake.

**Comment by pdcp1** [comment]
Last successful build https://app.netlify.com/sites/staging-aiid/deploys/652f08291989fd0d0d00da13

**Review by pdcp1 — APPROVED** [review]

**Pull request merged**